src/popup/pages/Settings.tsx

```diff
@@ -1,14 +1,36 @@
+import React from 'react';
 import { ConnectWalletForm } from '@/popup/components/ConnectWalletForm';
 import { WalletInformation } from '@/popup/components/WalletInformation';
-import { usePopupState } from '@/popup/lib/context';
-import React from 'react';
+import { useMessage, usePopupState } from '@/popup/lib/context';
+import { getWalletInformation } from '@/shared/helpers';
 
 export const Component = () => {
   const { state } = usePopupState();
+  const message = useMessage();
 
   if (state.connected) {
     return <WalletInformation info={state} />;
   } else {
-    return <ConnectWalletForm publicKey={state.publicKey} />;
+    return (
+      <ConnectWalletForm
+        publicKey={state.publicKey}
+        defaultValues={{
+          recurring:
+            localStorage?.getItem('connect.recurring') === 'true' || false,
+          amount: localStorage?.getItem('connect.amount') || undefined,
+          walletAddressUrl:
+            localStorage?.getItem('connect.walletAddressUrl') || undefined,
+        }}
+        onChange={(key, val) => {
+          localStorage?.setItem(`connect.${key}`, val.toString());
+        }}
+        getWalletInfo={getWalletInformation}
```

Comment:
Why do we pass `getWalletInformation` here instead of using it directly in the `ConnectWalletForm` component?